src/app/ReadClient.h

```diff
@@ -254,6 +261,23 @@ class ReadClient : public Messaging::ExchangeDelegate
     CHIP_ERROR AbortExistingExchangeContext();
     const char * GetStateStr() const;
 
+    /**
+     * Validate that the Event ID and Cluster ID in the header match that of the type information present in the object and
+     * decode the data. The template parameter T is generally expected to be a ClusterName::Events::EventName::Type struct
+     *
+     * @param [in] aEventHeader  The header of the event being validated.
+     * @param [in] aEvent        The event data.
+     * @param [in] aReader       The tlv reader.
+     */
+    template <typename EventDataT>
+    CHIP_ERROR DecodeEvent(const EventHeader & aEventHeader, const EventDataT & aEvent, TLV::TLVReader & aReader)
```

Comment:
This function is unused.  And comparing it to `TypedReadEventCallback:OnEventData` it returns a different error code for the schema mismatch.

What we should do is either remove this function or have `TypedReadEventCallback::OnEventData` call it.